CAF: add option to synchronize RPA rotation with advertising data updates (#10066)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

  * The :ref:`CONFIG_DESKTOP_BLE_DONGLE_PEER_ID_INFO <config_desktop_app_options>` configuration option.
    It can be used to indicate the dongle peer identity with a dedicated event.

>   * Updated the :ref:`nrf_desktop_fast_pair_app` to remove the Fast Pair advertising payload for the dongle peer.

Reviewer: This sentence could be moved to the updated section.

>   * Removed Fast Pair discoverable advertising payload on Resolvable Private Address (RPA) rotation during discoverable advertising session.

Reviewer: We need to have a new section for removed and move this sentence and one mentioned in line no:245 underneath the Removed header.

>   * Enabled synchronization of the Resolvable Private Address (RPA) rotation with the advertising data update in the Fast Pair configurations using the :kconfig:option:`CONFIG_CAF_BLE_ADV_ROTATE_RPA` option.

Reviewer: ```suggestion
  * Synchronization of the Resolvable Private Address (RPA) rotation to enable with the advertising data update in the Fast Pair configurations using the :kconfig:option:`CONFIG_CAF_BLE_ADV_ROTATE_RPA` option.
```
We could move this to the Updated section.

Author: ```suggestion
  * Synchronization between the Resolvable Private Address (RPA) rotation and the advertising data update in the Fast Pair configurations using the :kconfig:option:`CONFIG_CAF_BLE_ADV_ROTATE_RPA` option.
```

    * Set the max compiled-in log level to ``warning`` for the Non-Volatile Storage (:kconfig:option:`CONFIG_NVS_LOG_LEVEL`).
    * Lowered a log level to ``debug`` for the ``Identity x created`` log in the :ref:`nrf_desktop_ble_bond`.

  * By default, nRF Desktop dongles use the Bluetooth appearance (:kconfig:option:`CONFIG_BT_DEVICE_APPEARANCE)` of keyboard.
    The new default configuration value improves consistency with the used HID boot interface.

>   * Updated the :ref:`nrf_desktop_fast_pair_app` to remove the Fast Pair advertising payload for the dongle peer.

Reviewer: ```suggestion
  * The :ref:`nrf_desktop_fast_pair_app` to remove the Fast Pair advertising payload for the dongle peer.
```
Sorry, missed this in my previous review

Author: fixed

* Removed:

>   * Removed Fast Pair discoverable advertising payload on Resolvable Private Address (RPA) rotation during discoverable advertising session.
>   * Removed separate configurations enabling :ref:`zephyr:shell_api` (:file:`prj_shell.conf`).

Reviewer: ```suggestion
  * Fast Pair discoverable advertising payload on Resolvable Private Address (RPA) rotation during discoverable advertising session.
  * Separate configurations enabling :ref:`zephyr:shell_api` (:file:`prj_shell.conf`).
```

Author: Addressed. You can take a look at diff from the last force push:

https://github.com/nrfconnect/sdk-nrf/compare/6cf784ddb2d3587ba25ebd267cba074fc6c57431..9a71786ea31bc36ce3a7249d7252b98292393a42

* :ref:`bt_le_adv_prov_readme` library:

  * Extended the :c:struct:`bt_le_adv_prov_adv_state` structure to include new fields.
    The new :c:member:`bt_le_adv_prov_adv_state.rpa_rotated` field is used to notify registered providers about Resolvable Private Address (RPA) rotation.
    The new :c:member:`bt_le_adv_prov_adv_state.new_adv_session` field is used to notify registered providers that the new advertising session is about to start.
  * Changed the :kconfig:option:`CONFIG_BT_ADV_PROV_FAST_PAIR_BATTERY_DATA_MODE` Kconfig option (default value) to not include Fast Pair battery data in the Fast Pair advertising payload by default.

>   * Added the :kconfig:option:`CONFIG_BT_ADV_PROV_FAST_PAIR_STOP_DISCOVERABLE_ON_RPA_ROTATION` Kconfig option to drop the Fast Pair advertising payload on RPA rotation.

Reviewer: This could be moved to line no:526.
